tests/simplex/matrix_free_02.cc

```diff
@@ -0,0 +1,178 @@
+        for (unsigned int cell = cells.first; cell < cells.second; ++cell)
+          {
+            phi.reinit(cell);
+            for (unsigned int q = 0; q < phi.n_q_points; ++q)
+              phi.submit_value(1.0, q);
+
+            phi.integrate_scatter(true, false, dst);
+          }
+      },
+      vec,
+      dummy,
+      true);
+  }
+
+
+  void
+  vmult(VectorType &dst, const VectorType &src) const
+  {
+    matrix_free.template cell_loop<VectorType, VectorType>(
+      [&](const auto &, auto &dst, const auto &src, const auto cells) {
+        FEEvaluation<dim, -1, 0, 1, double> phi(matrix_free);
+        for (unsigned int cell = cells.first; cell < cells.second; ++cell)
+          {
+            phi.reinit(cell);
+            phi.gather_evaluate(src, false, true);
+
+            for (unsigned int q = 0; q < phi.n_q_points; ++q)
+              phi.submit_gradient(phi.get_gradient(q), q);
+
+            phi.integrate_scatter(false, true, dst);
+          }
+      },
+      dst,
+      src,
+      true);
+  }
+
+private:
+  const MatrixFree<dim, double> &matrix_free;
+};
+
+
+int
+main(int argc, char **argv)
+{
+  initlog();
+
+  Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+
+  const unsigned int dim    = 2; // only 2D is working for now
+  const unsigned int degree = 2;
+
+  // create mesh, select relevant FEM ingredients, and set up DoFHandler
+  Triangulation<dim> tria;
+
+  GridGenerator::subdivided_hyper_cube_with_simplices(tria, 16);
+
+  Simplex::FE_P<dim>   fe(degree);
+  Simplex::QGauss<dim> quad(degree + 1);
+  MappingFE<dim>       mapping(Simplex::FE_P<dim>(1));
+
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+
+  // Create constraint matrix
+  AffineConstraints<double> constraints;
+  VectorTools::interpolate_boundary_values(
+    mapping, dof_handler, 0, Functions::ZeroFunction<dim>(), constraints);
+  constraints.close();
+
+  // initialize MatrixFree
+  typename MatrixFree<dim, double>::AdditionalData additional_data;
+  additional_data.mapping_update_flags = update_gradients | update_values;
+
+  MatrixFree<dim, double> matrix_free;
+  matrix_free.reinit(mapping, dof_handler, constraints, quad, additional_data);
+
+  // create operator
+  PoissonOperator<dim> poisson_operator(matrix_free);
+
+  // initialize vectors
+  LinearAlgebra::distributed::Vector<double> x, b;
+  poisson_operator.initialize_dof_vector(x);
+  poisson_operator.initialize_dof_vector(b);
+
+  poisson_operator.rhs(b);
+
+  // solve linear equation system
+  ReductionControl                                     reduction_control;
+  SolverCG<LinearAlgebra::distributed::Vector<double>> solver(
+    reduction_control);
+  solver.solve(poisson_operator, x, b, PreconditionIdentity());
+
+  if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+    printf("Solved in %d iterations.\n", reduction_control.last_step());
+
+  constraints.distribute(x);
+
+#if false
```

Comment:
```suggestion
#if 0
```

I suspect (but I don't have evidence on hand) that some toolchains might print a warning with `#if false` so I'd prefer the unambiguously false version.